transport/server.cc

```diff
@@ -960,6 +960,8 @@ make_result(int16_t stream, messages::result_message& msg, const tracing::trace_
         cql_protocol_version_type version, bool skip_metadata = false);
 
 template<typename Process>
+requires std::same_as<std::invoke_result_t<Process, service::client_state&, distributed<cql3::query_processor>&, request_reader,
+        uint16_t, cql_protocol_version_type, service_permit, tracing::trace_state_ptr, bool, cql3::computed_function_values>, future<cql_server::process_fn_return_type>>
```

Comment:
Note, a real constraint should start with std::invocable, so that std::invoke_result_t is known to evaluate to something.

Reply:
Oh, I wasn't aware of this, I thought it's part of the concept.
But indeed this is the way it's defined in https://en.cppreference.com/w/cpp/types/result_of
```
the return type of the [Callable](https://en.cppreference.com/w/cpp/named_req/Callable) type F if invoked with the arguments ArgTypes.... Only defined if F can be called with the arguments ArgTypes... in unevaluated context.(since C++14)
```